CodingTracker.Mo3ses/Controller/CodingSessionController.cs

```diff
@@ -0,0 +1,70 @@
+using CodingTracker.Mo3ses.Interface;
+using CodingTracker.Mo3ses.Models;
+
+namespace CodingTracker.Mo3ses.Controller
+{
+    public class CodingSessionController
+    {
+        private readonly ICodingSessionRepository _codingSessionRepo;
+
+        public CodingSessionController(ICodingSessionRepository codingSessionRepo)
+        {
+            _codingSessionRepo = codingSessionRepo;
+        }
+
+        public void Create(string startTime, string endTime)
+        {
+            DateTime startTimeParsed = DateTime.Parse(startTime);
+            DateTime endTimeParsed = DateTime.Parse(endTime);
+
+            CodingSession codingSession = new CodingSession();
+            codingSession.StartTime = startTimeParsed;
+            codingSession.EndTime = endTimeParsed;
+
+            _codingSessionRepo.Create(codingSession);
+        }
+        public void Create(CodingSession codingSession)
+        {
+            _codingSessionRepo.Create(codingSession);
+        }
+
+        public void Delete(int codingSessionId)
+        {
+            CodingSession codingSession = new();
+            codingSession.Id = codingSessionId;
+
+            _codingSessionRepo.Delete(codingSession);
+        }
+
+        public List<CodingSession> GetAll()
+        {
+            return _codingSessionRepo.GetAll();
+        }
+
+        public void Update(int id, string startTime, string endTime)
+        {
```

Comment:
At the moment I'm not able to update a record, not sure why.
Also, when I input an end date before the start date, it's not allowed, but I don't see a message explaining what I need to do. It would be good to display an error message.

Reply:
I'll try to adjust everything today, I recently got a job as a helpdesk attendant and I'm still organizing it but by the end of the week I'll finish this project thank you for replying